util/src/trie/triedb.rs

```diff
@@ -293,7 +292,17 @@ impl<'a> TrieDBIterator<'a> {
 	/// The present key.
 	fn key(&self) -> Bytes {
 		// collapse the key_nibbles down to bytes.
-		self.key_nibbles.iter().step(2).zip(self.key_nibbles.iter().skip(1).step(2)).map(|(h, l)| h * 16 + l).collect()
+		unsafe {
```

Comment:
maybe the checked indexing is too slow? i would guess that using unsafe only for `get_unchecked` would be good enough.